### DISCUSSION TITLE
Why was `brew diy` removed?

### DISCUSSION THREAD

**Discussion by J-M0** [Homebrew/discussions#468] [Everyday usage]
I'm curious as to why the `diy` command was removed. I've tried searching issues and PRs, but I can't find any discussion about its removal.

**Comment by MikeMcQuaid** [comment]
It was broken a few times by other changes (it's the only place left that assumed a `Keg` could come from not-a-`Formula`), wasn't widely used and wasn't something we wished to continue to support.

**Reply by J-M0** [reply]
That's a bit disappointing as I liked `brew diy` better than `stow`. Is there any guidance for using `stow` and `brew` together?

**Reply by gromgit** [reply]
> That's a bit disappointing as I liked `brew diy` better than `stow`.

`brew diy` just ~tells~ told you what your package's Homebrew build prefix would be. `stow` actually links your package in, so a more appropriate comparison would be with `brew link`.

> Is there any guidance for using `stow` and `brew` together?

There's really only one: **Use different roots for each system** (the default `/usr/local` for Homebrew, and something else for Stow). This is "best practice" regardless of OS or packaging systems being considered, as trying to "cheat" one packaging system with another's product may work in the short term, but will likely cause weird issues in the longer term as each one starts interfering in the other's scope.

I use three different package managers on all my Ubuntu boxes, none of them share package roots, and so I've never had to clean up any messes:
* System (`apt`): `/usr` (logical equivalent)
* Homebrew: `/home/linuxbrew/.linuxbrew`
* [graft](http://peters.gormand.com.au/Home/tools/graft/): `/home/graft`
